### ISSUE TITLE
NullPointerException in extraction phase

### ISSUE THREAD

**Issue by dragos** [sbt/sbt#1593]
It seems the extraction phase is trying to be careful, but still calls `isDefined` on a `null` value.

```
2014-09-12 11:58:50,859 DEBUG [Worker-5] - EclipseSbtBuildManager - Calling Scala compiler with arguments  (CompilerInterface):
    -javabootclasspath
    /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Classes/classes.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Classes/ui.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Classes/jsse.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Classes/jce.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Classes/charsets.jar:/System/Library/Java/Extensions/AppleScriptEngine.jar:/System/Library/Java/Extensions/dns_sd.jar:/System/Library/Java/Extensions/j3daudio.jar:/System/Library/Java/Extensions/j3dcore.jar:/System/Library/Java/Extensions/j3dutils.jar:/System/Library/Java/Extensions/jai_codec.jar:/System/Library/Java/Extensions/jai_core.jar:/System/Library/Java/Extensions/mlibwrapper_jai.jar:/System/Library/Java/Extensions/MRJToolkit.jar:/System/Library/Java/Extensions/QTJava.zip:/System/Library/Java/Extensions/vecmath.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/lib/ext/apple_provider.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/lib/ext/dnsns.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/lib/ext/localedata.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/lib/ext/sunjce_provider.jar:/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/lib/ext/sunpkcs11.jar
    -javaextdirs

    -bootclasspath
    /Users/dragos/.m2/repository/p2/osgi/bundle/org.scala-lang.scala-library/2.11.3.v20140910-183426-7c8eaef41c/org.scala-lang.scala-library-2.11.3.v20140910-183426-7c8eaef41c.jar
    -deprecation
    -encoding
    UTF-8
    -classpath
    /Volumes/Thunderbolt_SSD/dragos/workspace/plugin/dragos-scala-ide/org.scala-ide.sdt.core.tests/target/work/data/builder-deprecation-warnings/bin:/Users/dragos/.m2/repository/p2/osgi/bundle/org.scala-lang.scala-reflect/2.11.3.v20140910-183426-7c8eaef41c/org.scala-lang.scala-reflect-2.11.3.v20140910-183426-7c8eaef41c.jar:/Users/dragos/.m2/repository/p2/osgi/bundle/org.scala-lang.scala-actors/2.11.3.v20140910-183426-7c8eaef41c/org.scala-lang.scala-actors-2.11.3.v20140910-183426-7c8eaef41c.jar
2014-09-12 11:58:50,900  INFO [Java indexing] - ScalaSourceIndexer - Indexing document: /builder-deprecation-warnings/src/test/B.scala
2014-09-12 11:58:50,952  INFO [Java indexing] - ScalaSourceIndexer - Indexing document: /builder-deprecation-warnings/src/test/B.scala
2014-09-12 11:58:50,962 ERROR [Worker-5] - org.scala-ide.sdt.core - org.scala-ide.sdt.core - org.scala-ide.sdt.core - 0 - Error in Scala compiler
java.lang.NullPointerException
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser$$anonfun$1.isDefinedAt(Dependency.scala:142)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser$$anonfun$1.isDefinedAt(Dependency.scala:141)
    at xsbt.Dependency$CollectTypeTraverser.traverse(Dependency.scala:99)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:144)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:114)
    at scala.reflect.api.Trees$Traverser.traverseTypeAscription(Trees.scala:2478)
    at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1207)
    at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
    at scala.reflect.internal.Trees$class.traverseMemberDef$1(Trees.scala:1196)
    at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1321)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:151)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:114)
    at scala.reflect.api.Trees$Traverser.traverseTrees(Trees.scala:2484)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:148)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:114)
    at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1200)
    at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
    at scala.reflect.internal.Trees$class.traverseMemberDef$1(Trees.scala:1196)
    at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1321)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:151)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:114)
    at scala.reflect.api.Trees$Traverser$$anonfun$traverseStats$1$$anonfun$apply$1.apply$mcV$sp(Trees.scala:2498)
    at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
    at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
    at scala.reflect.internal.Trees$class.itraverse(Trees.scala:1319)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
    at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:151)
    at xsbt.Dependency.xsbt$Dependency$$extractDependenciesByMemberRef(Dependency.scala:157)
    at xsbt.Dependency$DependencyPhase$$anonfun$run$2.apply(Dependency.scala:47)
    at xsbt.Dependency$DependencyPhase$$anonfun$run$2.apply(Dependency.scala:42)
    at scala.collection.Iterator$class.foreach(Iterator.scala:743)
    at scala.collection.AbstractIterator.foreach(Iterator.scala:1195)
    at xsbt.Dependency$DependencyPhase.run(Dependency.scala:42)
    at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1338)
    at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1325)
    at scala.tools.nsc.Global$Run.compileSources(Global.scala:1320)
    at scala.tools.nsc.Global$Run.compile(Global.scala:1418)
    at xsbt.CachedCompiler0.run(CompilerInterface.scala:123)
    at xsbt.CachedCompiler0.run(CompilerInterface.scala:99)
    at xsbt.CompilerInterface.run(CompilerInterface.scala:27)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:102)
    at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:48)
    at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:41)
    at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply$mcV$sp(AggressiveCompile.scala:99)
    at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:99)
    at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:99)
    at sbt.compiler.AggressiveCompile.sbt$compiler$AggressiveCompile$$timed(AggressiveCompile.scala:166)
    at sbt.compiler.AggressiveCompile$$anonfun$3.compileScala$1(AggressiveCompile.scala:98)
    at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:143)
    at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:87)
    at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:39)
    at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:37)
    at sbt.inc.IncrementalCommon.cycle(Incremental.scala:99)
    at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:38)
    at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:37)
    at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:65)
    at sbt.inc.Incremental$.compile(Incremental.scala:37)
    at sbt.inc.IncrementalCompile$.apply(Compile.scala:27)
    at sbt.compiler.AggressiveCompile.compile2(AggressiveCompile.scala:157)
    at sbt.compiler.AggressiveCompile.compile1(AggressiveCompile.scala:71)
    at sbt.compiler.AggressiveCompile.apply(AggressiveCompile.scala:46)
    at org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager.aggressiveCompile(EclipseSbtBuildManager.scala:215)
    at org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager.runCompiler(EclipseSbtBuildManager.scala:148)
    at org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager.update(EclipseSbtBuildManager.scala:134)
    at org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager.build(EclipseSbtBuildManager.scala:95)
    at org.scalaide.core.internal.project.ScalaProject.build(ScalaProject.scala:551)
    at org.scalaide.core.internal.builder.ScalaBuilder.build(ScalaBuilder.scala:127)
    at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
    at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:299)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:302)
    at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:358)
    at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:381)
    at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:143)
    at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:241)
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
```

**Comment by gkossakowski** [comment]
Here's the offending code:

``` scala
val typeSymbolCollector = new CollectTypeTraverser({
  case tpe if !tpe.typeSymbol.isPackage => tpe.typeSymbol
})
```

My guess would be that `typeSymbol` returns null. I don't know when that would happen. Do you have a sample code which reproduces this?

**Comment by dragos** [comment]
The annoying thing is that it won't happen every time. It's part of our test suite, the example is pretty small. My guess is that the `typesymbol` in question is the annotation.

```
// test/A.scala
package test

object A {
  val b = new B()
  println(b.a)
}
```

and

```
// test/B.scala
package test

class B {
  @deprecated
  def a = 2

  def c = 2
}
```

**Comment by gkossakowski** [comment]
Thanks! Do you have a I feel how often does it happen? 10% of the time or less often?

**Comment by jpsacha** [comment]
I am getting something similar that is happening for me 100% of time on certain projects using sbt 0.13.6 ans Scala 2.11.2. The stack trace is a bit different (see line numbers, here `Dependency.scala:132` above `Dependency.scala:142`):

```
java.lang.NullPointerException
        at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser$$anonfun$1.isDefinedAt(Dependency.scala:132)
        at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser$$anonfun$1.isDefinedAt(Dependency.scala:131)
        at xsbt.Dependency$CollectTypeTraverser.traverse(Dependency.scala:89)
        at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:134)
        at xsbt.Dependency$ExtractDependenciesByMemberRefTraverser.traverse(Dependency.scala:104)
        at scala.reflect.api.Trees$Traverser.traverseTypeAscription(Trees.scala:2478)
        at scala.reflect.internal.Trees$$anonfun$traverseMemberDef$1$1.apply$mcV$sp(Trees.scala:1207)
...
```

It there some workaround or setting to prevent this, short of moving back to 0.13.5?

**Comment by gkossakowski** [comment]
@jpsacha, thanks for the report! This looks indeed like the same problem. Would it be difficult for you to minimize reproduction of this problem or share your project?

The workaround is to switch back to old incremental compilation algorithm with `incOptions := incOptions.value.withNameHashing(false)`.

**Comment by jpsacha** [comment]
@gkossakowski I am not sure how small I can get it. I found a some smaller open-source projects where it is happening too every time I try. Though it is in a part that I did not publish yet, it may be a day or two due before it is on github. I will provide link when update it ready.

**Comment by gkossakowski** [comment]
Sounds good! It doesn't have to be too small as long as it's easy to reproduce.

**Comment by jpsacha** [comment]
Branch `SBT1593` of [javacv-examples](https://github.com/bytedeco/javacv-examples/tree/SBT1593) project has a build script that illustrates the issue. The example project is in subdirectory `FlyCapture2-demo`. To reproduce the issue, change to subdirectory `FlyCapture2-demo`, and run `sbt example_ui/run`. You need to do it on a Windows system with Java 8.0_20 (I tested on Windows 7 x64 and Windows 8 x64). It may work on Linux, but not other systems due to platform dependencies required by the build (only available for Windows and Linux).

Note that application itself (example_ui/FlyCaptureSnap) will crash on startup (to run it you will need to have newer version of flycature libraries 9.1-SNAPSHOT, not published yet, as on `develop` branch). Here is important that the build fails.

**Comment by dragos** [comment]
New report via @patriknw:

```
      while compiling: 
      /Users/patrik/dev/akka/akka-cluster/src/test/scala/akka/cluster/routing/WeightedRouteesSpec.scala
       during phase: xsbt-dependency
       library version: version 2.10.4
       compiler version: version 2.10.4
       reconstructed args: -bootclasspath /opt/eclipse-scala-luna/plugins/org.scala-ide.scala210.jars_4.0.0.201410211156/target/jars/scala-library.jar -Ywarn-nullary-unit -javabootclasspath /Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/rt.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/jfr.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/JObjC.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/dnsns.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/localedata.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/sunec.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/sunjce_provider.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/sunpkcs11.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/jre/lib/ext/zipfs.jar:/System/Library/Java/Extensions/AppleScriptEngine.jar:/System/Library/Java/Extensions/dns_sd.jar:/System/Library/Java/Extensions/j3daudio.jar:/System/Library/Java/Extensions/j3dcore.jar:/System/Library/Java/Extensions/j3dutils.jar:/System/Library/Java/Extensions/jai_codec.jar:/System/Library/Java/Extensions/jai_core.jar:/System/Library/Java/Extensions/mlibwrapper_jai.jar:/System/Library/Java/Extensions/MRJToolkit.jar:/System/Library/Java/Extensions/QTJava.zip:/System/Library/Java/Extensions/vecmath.jar -feature -Xlint -Ywarn-inaccessible -classpath /Users/patrik/dev/akka/akka-cluster/.target:/Users/patrik/dev/akka/akka-cluster/bin:/opt/eclipse-scala-luna/plugins/org.scala-ide.scala210.jars_4.0.0.201410211156/target/jars/scala-reflect.jar:/opt/eclipse-scala-luna/plugins/org.scala-ide.scala210.jars_4.0.0.201410211156/target/jars/scala-actor.jar:/opt/eclipse-scala-luna/plugins/org.scala-ide.scala210.jars_4.0.0.201410211156/target/jars/scala-swing.jar:/Users/patrik/.ivy2/cache/com.typesafe/config/bundles/config-1.2.0.jar:/Users/patrik/.ivy2/cache/io.netty/netty/bundles/netty-3.8.0.Final.jar:/Users/patrik/.ivy2/cache/com.google.protobuf/protobuf-java/bundles/protobuf-java-2.5.0.jar:/Users/patrik/.ivy2/cache/org.uncommons.maths/uncommons-maths/jars/uncommons-maths-1.2.2a.jar:/Users/patrik/.ivy2/cache/junit/junit/jars/junit-4.11.jar:/Users/patrik/.ivy2/cache/org.hamcrest/hamcrest-core/jars/hamcrest-core-1.3.jar:/Users/patrik/.ivy2/cache/org.scalatest/scalatest_2.10/jars/scalatest_2.10-2.1.3.jar:/Users/patrik/.ivy2/cache/com.codahale.metrics/metrics-core/bundles/metrics-core-3.0.1.jar:/Users/patrik/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.7.5.jar:/Users/patrik/.ivy2/cache/com.codahale.metrics/metrics-jvm/bundles/metrics-jvm-3.0.1.jar:/Users/patrik/.ivy2/cache/com.codahale.metrics/metrics-graphite/bundles/metrics-graphite-3.0.1.jar:/Users/patrik/.ivy2/cache/org.latencyutils/LatencyUtils/jars/LatencyUtils-1.0.3.jar:/Users/patrik/.ivy2/cache/org.hdrhistogram/HdrHistogram/jars/HdrHistogram-1.1.4.jar:/Users/patrik/.ivy2/cache/commons-codec/commons-codec/jars/commons-codec-1.7.jar:/Users/patrik/.ivy2/cache/org.apache.commons/commons-math/jars/commons-math-2.1.jar:/Users/patrik/.ivy2/cache/org.mockito/mockito-all/jars/mockito-all-1.8.1.jar:/Users/patrik/.ivy2/cache/org.scalacheck/scalacheck_2.10/jars/scalacheck_2.10-1.11.3.jar:/Users/patrik/.ivy2/cache/org.scala-sbt/test-interface/jars/test-interface-1.0.jar:/Users/patrik/.ivy2/cache/com.novocode/junit-interface/jars/junit-interface-0.10.jar:/Users/patrik/.ivy2/cache/junit/junit-dep/jars/junit-dep-4.10.jar:/Users/patrik/.ivy2/cache/org.scala-tools.testing/test-interface/jars/test-interface-0.5.jar:/Users/patrik/dev/akka/akka-remote/.target:/Users/patrik/dev/akka/akka-actor/.target:/Users/patrik/dev/akka/akka-actor-tests/.target:/Users/patrik/dev/akka/akka-testkit/.target:/Users/patrik/dev/akka/akka-remote-tests/.target:/Users/patrik/dev/akka/akka-multi-node-testkit/.target -unchecked -deprecation -Xlog-reflective-calls -Ypresentation-delay 200 -Ywarn-nullary-override -Ywarn-adapted-args -javaextdirs
            last tree to typer: TypeTree(class ActorRefRoutee)
            symbol: class ActorRefRoutee in package routing (flags: case)
            symbol definition: case class ActorRefRoutee extends Routee with Product with Serializable
            tpe: akka.routing.ActorRefRoutee
            symbol owners: class ActorRefRoutee -> package routing
            context owners: value testActorRoutee -> class WeightedRouteesSpec -> package routing 
            == Enclosing template or block ==  
            Template(   "AkkaSpec" 
                // parents   ValDef(     private     "_"     <tpt>     <empty>   )
                // 11 statements
                DefDef( // def <init>(): akka.cluster.routing.WeightedRouteesSpec in class WeightedRouteesSpec
                <method>     "<init>"     []     List(Nil)     <tpt> // tree.tpe=akka.cluster.routing.WeightedRouteesSpec     Block(       Apply(         super."<init>"         Apply(           "ConfigFactory"."parseString"           "\n      akka.actor.provider = \"akka.cluster.ClusterActorRefProvider\"\n      akka.remote.netty.tcp.port = 0\n      "         )       )       ()     )   )   ValDef( // private[this] val a1: akka.actor.Address in class WeightedRouteesSpec     private <local>     "a1"     <tpt> // tree.tpe=akka.actor.Address     Apply(       "Address"       // 4 arguments       "akka.tcp"       "sys"       "a1"       2551     )   )   ValDef( // private[this] val b1: akka.actor.Address in class WeightedRouteesSpec     private <local>     "b1"     <tpt> // tree.tpe=akka.actor.Address     Apply(       "Address"       // 4 arguments       "akka.tcp"       "sys"       "b1"       2551     )   )   ValDef( // private[this] val c1: akka.actor.Address in class WeightedRouteesSpec     private <local>     "c1"     <tpt> // tree.tpe=akka.actor.Address     Apply(       "Address"       // 4 arguments       "akka.tcp"       "sys"       "c1"       2551     )   )   ValDef( // private[this] val d1: akka.actor.Address in class WeightedRouteesSpec     private <local>     "d1"     <tpt> // tree.tpe=akka.actor.Address     Apply(       "Address"       // 4 arguments       "akka.tcp"       "sys"       "d1"       2551     )   )   ValDef( // private[this] val routeeA: akka.routing.ActorSelectionRoutee in class WeightedRouteesSpec     private <local>     "routeeA"     <tpt> // tree.tpe=akka.routing.ActorSelectionRoutee     Apply(       "ActorSelectionRoutee"       Apply(         "system"."actorSelection"         Apply(           RootActorPath(a1).$div("user")."$div"           "a"         )       )     )   )   ValDef( // private[this] val routeeB: akka.routing.ActorSelectionRoutee in class WeightedRouteesSpec     private <local>     "routeeB"     <tpt> // tree.tpe=akka.routing.ActorSelectionRoutee     Apply(       "ActorSelectionRoutee"       Apply(         "system"."actorSelection"         Apply(           RootActorPath(b1).$div("user")."$div"           "b"         )       )     )   )   ValDef( // private[this] val routeeC: akka.routing.ActorSelectionRoutee in class WeightedRouteesSpec     private <local>     "routeeC"     <tpt> // tree.tpe=akka.routing.ActorSelectionRoutee     Apply(       "ActorSelectionRoutee"       Apply(         "system"."actorSelection"         Apply(           RootActorPath(c1).$div("user")."$div"           "c"         )       )     )   )   ValDef( // private[this] val routees: scala.collection.immutable.Vector[akka.routing.ActorSelectionRoutee] in class WeightedRouteesSp
```

**Comment by dragos** [comment]
@gkossakowski This seems to be more common that initially thought. We need to fix this at some level.

**Comment by patriknw** [comment]
I get this a lot now in Eclipse with latest 4.0 RC1, in various projects. Also when running Eclipse with JDK 1.7

**Comment by dragos** [comment]
@patriknw can you check if name hashing is enabled? If yes, I think disabling it should work around this bug.

**Comment by dragos** [comment]
Info gathered yesterday during a debugging session:

The NPE comes from a `TypeTree` with its `tpe` being `null`. This `TypeTree` is the `tpt` of a `DefDef` of an object constructor. My guess is that that particular type tree is never inspected in the backend, since it's not useful: Scala pretends the return type of a constructor is the enclosing class itself, while at the JVM level constructors always return `void`.

This seems to be related to the fact that Sbt compiles sources for which there are already classfiles in the output folder. Couldn't reproduce it outside of the IDE for the moment.

**Comment by patriknw** [comment]
I used name hashing. Disabled that now. Right now no errors, but no class files in output directory, which is even worse.

**Comment by gkossakowski** [comment]
I checked implementation of `TypeTree` in a compiler. From what I saw the `tpe` should never be null after typechecking so the NPE here is a bug in Scala compiler.

However, given we have a hard time reproducing it (and understanding it) then I think inserting an explicit null check is a good workaround. Still, the best would be to track down the root cause of the problem. We've done this in the past by emitting a warning and asking users to report back if they find reliable way of triggering a warning. See cb0e723923c7dd91f2cf1aa8edc2996635dd5f8d for example.
